### ISSUE TITLE
[feature request] Improved sharing capabilities

### ISSUE THREAD

**Issue by ITwrx** [owncloud/core#14348]
I'm new to owncloud so please correct me if i'm missing something, but it seems there is no way to define complex sharing priveleges/restrictions. It seems to me that if i want a user/group to be able to share with the admin(or other user/group) but noone else, there is no way to do this except to create a unique group for each user and add the admin(or other sharee) to that group. That doesn't scale well at all and would confuse the end user significantly. IMO, it should be possible to define who can share with who and in what direction this sharing can occur. It weould be nice if user existence/presence info could be exposed in the same way but that should probably be a seperate FR. pls see the image below for how these sharing associaations/priveleges could be exposed to the admin.
![owncloud_sharing_rights](https://cloud.githubusercontent.com/assets/1002762/6252384/8cb7d390-b75f-11e4-8fc2-c87095c70f7e.jpg)

thanks in advance

**Comment by jancborchardt** [comment]
It would be cool if you can describe your specific use case. As the solution seems very difficult to understand and also difficult to scale.

**Comment by andriesinfoserv** [comment]
sure, let me try again.

note: it turns out that the current sharing perms are more flexible than i first realized but there still seem to be some limitations. namely, which direction the sharing can occur and single user to single user sharing and which direction. 

For example, beebo bread(client) can share with admin bidirectionally by creating a specific group for beebo and adding admin to beebo grp. we can ensure beebo can't share our invoices with mrs. baird's by setting the "share with own group" setting and we can allow photography group to share bread pics with beebo grp by adding photography to beebo grp but now beebo can share our invoices with photography and photography can turn around and share those invoices with mrs. baird's, assuming photgraphy's relationship with mrs. baird's is the same as it's relationship with beebo. If we could specify direction of sharing we could say that photography can share bread pics with beebo but beebo can't share invoices with photography. 

maybe i'm just missing a simple way to implement this using the existing sharing capabilities?

Also, i can imagine that having the ability to specify sharing relationships and direction for single users would reduce config complexity and improve capabilities in certain situations, as one could define the exact relationship they were after instead of creating workarounds using groups or possibly making privacy/security concessions. If you would like me to try and construct a use case for that too, let me know.
Currently my primary concern is illustrated by the example above. i'm just guessing that down the line the user to user capability would be an issue too. maybe not.

When i spoke of scaling i was only referring to the end user labor of creating seperate groups for each client(for instance) just to enable bidirectional sharing with the admin. Creating these groups may be necessary regardless, depending on end user needs and how this all turns out, but adding sharing direction would add return on that labor investment.
As far as the scaling of the feature in owncloud(server resources, processing time, etc): i'm sure that's a valid concern. I just think that sharing capabilities are central to owncloud's purpose and deserve a close look. 

Thanks again.

**Comment by andriesinfoserv** [comment]
also. two way sharing between all members of a staff group(for instance) with admin user but not each other. right now you have to create a seperate group for each user to be able to get two way sharing working without also allowing those users to share amounst themselves. This is the scenario i tried to illustrate in the first post. 

sometimes, users belong in a group and sharing is ok between all of them. sometimes they need to be in a group so they can share with another group or user or so that that group or user can share with them, not so they can share with themselves.

hope that's clearer.

**Comment by jancborchardt** [comment]
Sharing is already complicated enough, and on top of it there seems to be no one who has the time to work on it. So I’ll close this.
